src/fluree/db/query/fql/parse.cljc

```diff
@@ -437,14 +443,25 @@
 
 (defn parse-object-map
   [s-mch p-mch o vars context]
-  (let [o* (expand-keys o context)]
-    (if-let [v (get o* const/iri-value)]
-      (let [attrs (dissoc o* const/iri-value)
-            o-mch (if-let [var (parse-var-name v)]
-                    (parse-variable-attributes var attrs vars context)
+  (let [o* (expand-keys o context)
+        explicit-var (get o "@variable")]
+    (cond
+      explicit-var
+      (let [attrs (-> o (dissoc "@variable") (expand-keys context))
+            var   (parse-var-name explicit-var)
+            o-mch (parse-variable-attributes var attrs vars context)]
+        [(flip-reverse-pattern [s-mch p-mch o-mch])])
+
+      (contains? o* const/iri-value)
+      (let [v     (get o* const/iri-value)
+            attrs (dissoc o* const/iri-value)
+            o-mch (if (and (parse-var-name v)
+                           *object-var-parsing*)
```

Comment:
I think we should check `*object-var-parsing*` first, since it's less work than `parse-var-name`.